tests/unit/test_resolver.py

```diff
@@ -93,3 +97,71 @@ def fetch(req: Request) -> RequestOutput:
         assert resolver.resolve_packages(request) == COMBINED_OUTPUT
 
     assert calls_by_pkgtype == ["gomod", "npm", "pip"]
+
+
+@pytest.mark.parametrize(
+    "flags",
+    [
+        pytest.param(["dev-mode"], id="dev-mode-true"),
+        pytest.param([], id="dev-mode-false"),
+    ],
+)
+def test_dev_mode(flags: list[str]) -> None:
+    with mock.patch.dict(
+        resolver._dev_package_managers,
+        values={"shrubbery": lambda _: RequestOutput.empty()},
+        clear=True,
+    ):
+
+        class foo:
+            type: str
+
+        dev_package_input = foo()
+        dev_package_input.type = "shrubbery"
+
+        request = mock.Mock()
+        request.flags = flags
+        request.packages = [dev_package_input]
+
+        if flags:
+            assert resolver.resolve_packages(request) == RequestOutput(
+                components=[], build_config=BuildConfig(environment_variables=[], project_files=[])
+            )
+        else:
+            expected_error = re.escape("Package manager(s) not yet supported: shrubbery")
+            try:
+                resolver.resolve_packages(request)
+            except UnsupportedFeature as exc:
+                assert re.match(expected_error, str(exc))
+
+
+def test_dev_mode_with_released() -> None:
+    with (
+        mock.patch.dict(
+            resolver._package_managers,
+            values={"gomod": lambda _: RequestOutput.empty()},
+            clear=True,
+        ),
+        mock.patch.dict(
+            resolver._dev_package_managers,
+            values={"pip": lambda _: RequestOutput.empty()},
+            clear=True,
+        ),
+    ):
+
+        class foo:
+            type: str
+
+        dev_package_input = foo()
+        dev_package_input.type = "pip"
+
+        released_package_input = mock.Mock()
+        released_package_input.type = "gomod"
+
+        request = mock.Mock()
+        request.flags = ["dev-mode"]
+        request.packages = [released_package_input, dev_package_input]
+
+        assert resolver.resolve_packages(request) == RequestOutput(
```

Comment:
I believe we should add more specific test for resolved packages to see that we actually resolved both released and dev packages. In case that I would remove `released_package_input` from packages now I believe test would still pass.

Reply:
We could probably mock the resolver functions and then assert they were called 2x:
```
mock_resolver = mock.Mock()
mock_resolver.return_value = RequestOutput.empty()

    with (
        mock.patch.dict(
            resolver._package_managers,
            values={"gomod":mock_resolver},
            clear=True,
        ),
        mock.patch.dict(
            resolver._dev_package_managers,
            values={"pip": mock_resolver},
            clear=True,
        ),
    ):
    # ...
    assert mock_resolver.call_count == 2
```